tests/test_routes.py

```diff
@@ -0,0 +1,211 @@
+
+def test_create_one_planet(client):
+
+    response = client.post("/planets", json={
+        "name": "Pluto",
+        "description": "A real planet!",
+        "moons": 0,
+        "life": False 
+    })
+    response_body = response.get_json()
+
+    assert response.status_code == 201
+    assert response_body == {
+        "id":1,
+        "name": "Pluto",
+        "description": "A real planet!",
+        "moons": 0,
+        "life": False 
+    }
+
+def test_create_planet_with_a_missing_key(client):
+
+    response = client.post("/planets", json={
+        "description": "A big blue planet!",
+        "moons": 1,
+        "life": True 
+    })
+
+    response_body = response.get_json()
+
+    assert response.status_code == 400
+    assert response_body == {"details":"Missing key: 'name'"}
+
+
+def test_delete_planet(client, two_planets):
+
+    response = client.delete("/planets/1")
+    response_body = response.get_data(as_text=True)
+
+    assert response.status_code == 200
+    assert response_body == '*** You have successfully destroyed Earth ! ***'
+
+
+def test_get_planet_id_not_found(client):
+
+    response = client.get("/planets/1")
+    response_body = response.get_json()
+
+    assert response.status_code == 404
+    assert response_body == {'details': 'No planet with id 1 found'}
+
+
+
+def test_update_planet_description(client, two_planets):
+
+    response = client.patch("/planets/1", json={
+        "description": "A big blue planet!",
+    })
+
+    response_body = response.get_json()
+
+    assert response.status_code == 200
+    assert response_body == {
+        "id" : 1,
+        "name": "Earth",
+        "description": "A big blue planet!",
+        "moons": 1,
+        "life": True 
+    }
+
+def test_update_planet_name(client, two_planets):
+
+    response = client.patch("/planets/1", json={
+        "name": "Earth 2.0",
+    })
+
+    response_body = response.get_json()
+
+    assert response.status_code == 200
+    assert response_body == {
+        "id" : 1,
+        "name": "Earth 2.0",
+        "description": "home",
+        "moons": 1,
+        "life": True 
+    }
+
+def test_update_planet_life(client, two_planets):
+
+    response = client.patch("/planets/1", json={
+        "life": False,
+    })
+
+    response_body = response.get_json()
+
+    assert response.status_code == 200
+    assert response_body == {
+        "id" : 1,
+        "name": "Earth",
+        "description": "home",
+        "moons": 1,
+        "life": False
+    }
+
+def test_update_planet_moons(client, two_planets):
+
+    response = client.patch("/planets/1", json={
+        "moons":0,
+    })
+
+    response_body = response.get_json()
+
+    assert response.status_code == 200
+    assert response_body == {
+        "id" : 1,
+        "name": "Earth",
+        "description": "home",
+        "moons":0,
+        "life": True 
+    }
+
+def test_update_planet_with_a_put(client, two_planets):
```

Comment:
For a lot of these methods that modify the database. We might also want to make sure that our endpoint is actually saving the changes (we don't want to forget a commit).

The post endpoint is fine as is, since the record won't get an id unless it is committed.

But for the Put and PATCH endpoints, consider issue a follow-up get to the record id to ensure that the data is actually saved, and not simply returned from the endpoint without committing it.